tracker/labels.go

```diff
@@ -19,46 +21,65 @@ func GetTimestamp() string {
 }
 
 // ReplaceLabel replaces the detection with an almost identical detection (new label)
-func ReplaceLabel(det objdet.Detection, label string) objdet.Detection {
-	return objdet.NewDetection(*det.BoundingBox(), det.Score(), label)
+func ReplaceLabel(tr *track, label string) *track {
+	det := objdet.NewDetection(*tr.Det.BoundingBox(), tr.Det.Score(), label)
+	newTrack := tr.clone()
+	newTrack.Det = det
+	return newTrack
+}
+
+// ReplaceBoundingBox replaces the detection with an almost identical detection (new bounding box)
+func ReplaceBoundingBox(tr *track, bb *image.Rectangle) *track {
+	det := objdet.NewDetection(*bb, tr.Det.Score(), tr.Det.Label())
+	newTrack := tr.clone()
+	newTrack.Det = det
+	return newTrack
 }
 
 // RenameFromMatches takes the output of the Hungarian matching algorithm and
 // gives the new detection the same label as the matching old detection.  Any new detections
 // found will be given a new name (and cleass counter will be updated)
 // Also return freshDets that are the fresh detections that were not matched with any detections in the previous frame.
-func (t *myTracker) RenameFromMatches(matches []int, matchinMtx [][]float64, oldDets, newDets []objdet.Detection) ([]objdet.Detection, []objdet.Detection) {
+func (t *myTracker) RenameFromMatches(matches []int, matchinMtx [][]float64, oldDets, newDets []*track) ([]*track, []*track, []*track) {
```

Comment:
Biggest thing here is that it adds a new category, called "newlyStable", and also separates out the updated Tracks from the newly stable and fresh tracks